### PR TITLE
Wallet module

### PR DESCRIPTION
**Pull request by vamsitallam**

### CODE REVIEW

File: lib/wallet/components/SignInPage/index.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import {observer} from 'mobx-react-lite';
+
+import {WalletStore} from '../../stores/WalletStore/WalletStore';
+
+interface Props {
+  walletStore: WalletStore;
+}
+
+const SignInPage = observer((props: Props) => {
+  if (props.walletStore) {
```
Comment (PavanGangireddy) [inline lib/wallet/components/SignInPage/index.tsx:11]: is this if needed?

File: lib/wallet/stores/WalletStore/WalletStore.ts [hunk 2]
```diff
@@ -1,42 +1,131 @@
...
+      resetWalletData: action.bound,
+    });
+  }
+
+  setError(e) {
+    this.error = e;
+  }
+
+  setIsConnected(isConnected) {
+    this.isConnected = isConnected;
```
Comment (PavanGangireddy) [inline lib/wallet/stores/WalletStore/WalletStore.ts:47]: isConnected should be a computed value

File: lib/wallet/stores/WalletStore/WalletStore.ts [hunk 3]
```diff
@@ -1,42 +1,131 @@
...
+  }
+
+  resetWalletData() {
+    this.setIsConnected(false);
+    this.setIsConnecting(false);
+    this.setWalletData(null);
+  }
+
+  listenToWalletEvents(connection) {
+    // Subscribe to accounts change
```
Comment (PavanGangireddy) [inline lib/wallet/stores/WalletStore/WalletStore.ts:104]: Always add event listeners in components, its not store responsibilty

File: lib/wallet/utils/utils.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import {ethers} from 'ethers';
+
+import {connectionManager} from '../../connect';
```
Comment (PavanGangireddy) [inline lib/wallet/utils/utils.ts:3]: connectionManager name change to connection

File: lib/connect/connectionManager.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        this.web3Modal.cachedProvider,
+      );
+      const provider = new ethers.providers.Web3Provider(connection);
+
+      return provider;
+    }
+    return null;
+  };
+
+  getCachedConnection = async () => {
```
Comment (PavanGangireddy) [inline lib/connect/connectionManager.ts:31]: is this needed?

File: lib/connect/connectionManager.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    if (typeof window !== 'undefined') {
+      this.web3Modal = new Web3Modal({
+        network: 'mainnet',
+        cacheProvider: true,
+        providerOptions,
+      });
+    }
+  }
+
+  getConnectedProvider = async () => {
```
Comment (PavanGangireddy) [inline lib/connect/connectionManager.ts:19]: we can handle new connection in the else case here... so always use getConnectedProvider everywhere, remove getCachedConnection

File: lib/wallet/providers/WalletProvider/WalletProvider.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      setError(null);
+      if (!!provider) {
+        try {
+          await onConnect();
+        } catch (e) {
+          setError(e);
+        }
+      }
+    }
+    initializeWeb3ModalAndAddListeners();
```
Comment (PavanGangireddy) [inline lib/wallet/providers/WalletProvider/WalletProvider.tsx:24]: add event listeners here

File: src/constants/walletProviderConstants.ts [hunk 8]
```diff
@@ -9,16 +9,16 @@ import WalletLink from 'walletlink';
 const infuraId = 'b28b5d2dec194976b90c78dbd42cf649';
 
 export const providerOptions = {
```
Comment (PavanGangireddy) [inline src/constants/walletProviderConstants.ts:11]: is this not needed? remove file if not needed

File: src/context/walletStoreContext.tsx [hunk 9]
```diff
@@ -1,6 +1,21 @@
...
 export {WalletStoreContext};
+
+export type WalletProviderProps = FC<{
+  children: ReactNode;
+}>;
+
+export const WalletStoreProvider: WalletProviderProps = ({
+  children,
+}): ReactElement => {
+  const store = new WalletStore();
```
Comment (PavanGangireddy) [inline src/context/walletStoreContext.tsx:15]: Remove this and integrate with Root Store

File: src/pages/connect.tsx [hunk 10]
```diff
@@ -1,8 +1,13 @@
-import {ConnectWallet} from '@vamsitallam/web3-react';
+// import {ConnectWallet} from '@vamsitallam/web3-react';
 import {Flex, Text} from '@chakra-ui/react';
```
Comment (PavanGangireddy) [inline src/pages/connect.tsx:2]: is this package needed?

File: src/pages/connect.tsx [hunk 11]
```diff
@@ -32,15 +37,20 @@ export default function Connect() {
...
-      <ConnectWallet />
+      {walletStore.isConnected ? (
+        <Text mb="12px" fontWeight="bold">
+          Your Details:
+        </Text>
+      ) : null}
+      {/* <ConnectWallet /> */}
+      <SignIn walletStore={walletStore} />
 
       {/* <Text mt="48px" mb="12px" fontWeight="bold">
```
Comment (PavanGangireddy) [inline src/pages/connect.tsx:48]: remove commented code or add Note or todo or explaining it

File: src/pages/my-assets.tsx [hunk 12]
```diff
@@ -1,58 +1,64 @@
...
-  const {marketPlaceStore} = useStores();
+  const {marketPlaceStore, walletStore} = useStores();
+  const {isConnected, isConnecting, address} = walletStore;
 
   useEffect(() => {
-    setLoadingState('loading');
-    if (!loading) {
-      console.log(loading, address);
+    if (isConnecting) {
+      setLoadingState('connecting wallet');
```
Comment (PavanGangireddy) [inline src/pages/my-assets.tsx:22]: is this loading state needed? 
cant we use isConnecting here?

File: src/pages/my-assets.tsx [hunk 13]
```diff
@@ -1,58 +1,64 @@
...
-    if (address === undefined) {
-      router.push('/connect');
-      return;
-    }
+    setLoadingState('fetching my nfts');
+
     const data = await marketPlaceStore.fetchMyNFTs();
+    setLoadingState('resolving data');
+
+    if (nftContract) {
```
Comment (PavanGangireddy) [inline src/pages/my-assets.tsx:35]: if (nftContract)  - is this needed here? why?

File: src/utils/transactionUtils.ts [hunk 14]
```diff
@@ -1,8 +1,10 @@
-import {getConnectedProvider} from '@/lib/connect';
+// import {getConnectedProvider} from '@/lib/connect';
+import {getConnectedProvider} from '@/lib/wallet/utils/utils';
 import {ethers} from 'ethers';
 
 const sendTransaction = async (contractData, getTransaction) => {
   // TODO: Integrate provider and signer here
+  // const provider = await getConnectedProvider();
```
Comment (PavanGangireddy) [inline src/utils/transactionUtils.ts:7]: remove comments

Comment (PavanGangireddy) [reply src/utils/transactionUtils.ts:7]: lets add try catch block for entire thing

Comment (PavanGangireddy) [reply src/utils/transactionUtils.ts:7]: add try catch block inside getConnected function throw respective errors there

**Review by PavanGangireddy — COMMENTED** [review]
Added all review inputs

File: src/pages/my-assets.tsx (hunk 13 above)
Comment (vamsitallam) [reply src/pages/my-assets.tsx:35]: initially nftContarct is null

File: lib/wallet/components/SignInPage/index.tsx (hunk 1 above)
Comment (vamsitallam) [reply lib/wallet/components/SignInPage/index.tsx:11]: Done

File: lib/wallet/stores/WalletStore/WalletStore.ts (hunk 2 above)
Comment (vamsitallam) [reply lib/wallet/stores/WalletStore/WalletStore.ts:47]: Done

File: lib/connect/connectionManager.ts (hunk 5 above)
Comment (vamsitallam) [reply lib/connect/connectionManager.ts:31]: This util is used for establishing a connection to wallet incase of reloading

File: src/constants/walletProviderConstants.ts (hunk 8 above)
Comment (vamsitallam) [reply src/constants/walletProviderConstants.ts:11]: Done, Deleted

File: src/context/walletStoreContext.tsx (hunk 9 above)
Comment (vamsitallam) [reply src/context/walletStoreContext.tsx:15]: Done deleted the file, as we are not using it

File: src/pages/my-assets.tsx (hunk 12 above)
Comment (vamsitallam) [reply src/pages/my-assets.tsx:22]: I have kept it to understand and display different loading states in UI for ebugging

File: src/utils/transactionUtils.ts (hunk 14 above)
Comment (vamsitallam) [reply src/utils/transactionUtils.ts:7]: Done

File: src/pages/connect.tsx (hunk 10 above)
Comment (vamsitallam) [reply src/pages/connect.tsx:2]: Done Removed

File: src/pages/connect.tsx [hunk 15]
```diff
@@ -1,8 +1,13 @@
-import {ConnectWallet} from '@vamsitallam/web3-react';
+// import {ConnectWallet} from '@vamsitallam/web3-react';
 import {Flex, Text} from '@chakra-ui/react';
 import React from 'react';
+import SignIn from '../../lib/wallet/components/SignInPage';
```
[the remark this answers is not in the document]
Comment (vamsitallam) [reply src/pages/connect.tsx:4]: Done

File: src/pages/_app.tsx [hunk 16]
```diff
@@ -3,12 +3,13 @@ import React from 'react';
 import {ThirdwebProvider} from '@vamsitallam/web3-react';
```
[the remark this answers is not in the document]
Comment (vamsitallam) [reply src/pages/_app.tsx:3]: Done removed

File: src/pages/connect.tsx (hunk 11 above)
Comment (vamsitallam) [reply src/pages/connect.tsx]: Done, removed

File: lib/wallet/stores/WalletStore/WalletStore.ts (hunk 3 above)
Comment (vamsitallam) [reply lib/wallet/stores/WalletStore/WalletStore.ts:103]: Done, moved to wallet provider

File: lib/wallet/utils/utils.ts (hunk 4 above)
Comment (vamsitallam) [reply lib/wallet/utils/utils.ts:3]: Done

**Pull request merged**